src/transport_performance/gtfs/routes.py

```diff
@@ -131,3 +137,50 @@ def scrape_route_type_lookup(
     route_lookup = pd.DataFrame(zip(cds, txts), columns=["route_type", "desc"])
 
     return route_lookup
+
+
+def get_saved_route_type_lookup(
+    path: Union[str, pathlib.Path] = pathlib.Path(
+        "tests/data/gtfs/route_lookup.pkl"
+    )
+) -> pd.DataFrame:
+    """Get the lcoally saved route type lookup as a dataframe.
+
+    Parameters
+    ----------
+    path : Union[str, pathlib.Path], optional
+        The path to the route type lookup,
+        by default pathlib.Path("tests/data/gtfs/route_lookup.pkl")
+
+    Returns
+    -------
+    pd.DataFrame
+        The route type lookup
+
+    """
+    # defences
+    _is_expected_filetype(
+        pth=path, param_nm="path", check_existing=True, exp_ext=".pkl"
+    )
+    lookup = pd.read_pickle(path)
+    ACCEPTED_TYPES = (dict, pd.DataFrame)
+    # check unserialized .pkl file is of correct type
+    if not isinstance(lookup, ACCEPTED_TYPES):
+        raise TypeError(
+            "Serialized object in specified .pkl file is of type: "
+            f"{type(lookup)}. Expected {ACCEPTED_TYPES}"
+        )
+    # convert dicts to pandas df
+    if isinstance(lookup, dict):
```

Comment:
Why do we need to handle cases where the returned pickle object is a dict? Can we not rely on it to always be pd df on all operating systems? Also in test_routes.py , `# test a .pkl containing a serialized dict` , may be worth chatting me through this one.

Reply:
My general thoughts where that dict objects could be converted to dfs, that's why there's handling of them in here.
The veal though it may not end up having any use cases unless we make some additions in the future